test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -1765,6 +1828,224 @@ Describe "Invoke-RestMethod tests" -Tags "Feature" {
 
     #endregion Certificate Authentication Tests
 
+    #region charset encoding tests
+
+    Context  "Invoke-RestMethod Encoding tests with BasicHtmlWebResponseObject response" {
+        It "Verifies Invoke-RestMethod detects charset meta value when the ContentType header does not define it." {
+            $output = '<html><head><meta charset="Unicode"></head></html>'
+            $expectedEncoding = [System.Text.Encoding]::GetEncoding('Unicode')
+            $response = ExecuteRestMethod -Uri "http://localhost:8081/PowerShell?test=response&output=$output" -UseBasicParsing
+
+            $response.Error | Should BeNullOrEmpty
+            $response.Encoding.EncodingName | Should Be $expectedEncoding.EncodingName
+        }
+
+        It "Verifies Invoke-WebRequest detects charset meta value when newlines are encountered in the element." {
+            $output = @'
```

Comment:
just a question, would: 
```
"<html>`n<head>`n<meta`n   charset=""Unicode""  `n>`n</head>`n</html>`n"
```
be the same?

Reply:
Yes, but I find the HERE string much easier to read and it avoids me making silly escaping errors.